assembly/index.ts

```diff
@@ -0,0 +1,101 @@
+import { Request,  Response, Fastly, Headers } from "@fastly/as-compute";
+
+// The name of a backend server associated with this service.
+//
+// This should be changed to match the name of your own backend. See the the
+// `Hosts` section of the Fastly Wasm service UI for more information.
+const BACKEND_NAME = "backend_name";
+
+/// The name of a second backend associated with this service.
+const OTHER_BACKEND_NAME = "other_backend_name";
+
+function getQueryParam(qs: string, param: string): string {
+    const pairs = qs.split("&");
+    for (let i = 0; i < pairs.length; i++) {
+        if (pairs[i].indexOf(param + "=")==0) {
+            return pairs[i].substr(param.length + 1);
+        }
+    }
+    return "";
+}
+
+function getQueryString(path: string):string {
+    if (path.indexOf("?") > 0) {
+        return path.substring(path.indexOf("?") + 1);
+    }
+    return "";
+}
+
+// The entry point for your application.
+//
+// Use this function to define your main request handling logic. It could be
+// used to route based on the request properties (such as method or path), send
+// the request to a backend, make completely new requests, and/or generate
+// synthetic responses.
+function main(req: Request): Response {
+    // Make any desired changes to the client request.
+    req.headers().set("Host", "example.com");
+
+    // We can filter requests that have unexpected methods.
+    const VALID_METHODS = ["HEAD", "GET", "POST"];
+    if (!VALID_METHODS.includes(req.method())) {
+        return new Response(String.UTF8.encode("This method is not allowed"), {
+            status: 405,
+        });
+    }
+
+    let method = req.method();
+    let urlParts = req.url().split("//").pop().split("/");
+    let host = urlParts.shift();
+    let path = ("/" + urlParts.join("/"));
+    let qs = getQueryString(path);
+
+    const owner = getQueryParam(qs, "owner");
+    const repo = getQueryParam(qs, "repo"); 
+    let ref = getQueryParam(qs, "ref");
+    let sha = "";
+
+    if (owner != "" && repo != "" && true) {
```

Comment:
can we use the same code style rules also for typescript? eg single quotes etc.. ?